core/src/testUtils/groovy/nu/yona/server/test/AppService.groovy

```diff
@@ -295,14 +313,75 @@ class AppService extends Service
 		}
 	}
 
-	def assertResponseStatusCreated(def response)
+	void assertDefaultOwnDevice(def device)
+	{
+		if (device instanceof Device)
+		{
+			assert device.name == "First device"
+			assert device.operatingSystem == "UNKNOWN" 
+		}
+		else
+		{
+			assert device.keySet() == ["name", "operatingSystem", "registrationTime", "appLastOpenedDate", "vpnConnected", "_links"] as Set
+			assert device._links.keySet() == ["self", "edit"] as Set
+			assert device.name == "First device"
+			assert device.operatingSystem == "UNKNOWN"
+			assertDateTimeFormat(device.registrationTime)
+			assertDateFormat(device.appLastOpenedDate)
+			assert device.vpnConnected == true || device.vpnConnected == false
+		}
+	}
+
+	static void assertEquals(String dateTimeString, ZonedDateTime comparisonDateTime, int epsilonSeconds = 10)
+	{
+		// Example date/time string: 2016-02-23T21:28:58.556+0000
+		ZonedDateTime dateTime = YonaServer.parseIsoDateTimeString(dateTimeString)
+		assertEquals(dateTime, comparisonDateTime, epsilonSeconds)
+	}
+
+	static void assertEquals(ZonedDateTime dateTime, ZonedDateTime comparisonDateTime, int epsilonSeconds = 10)
+	{
+		int epsilonMilliseconds = epsilonSeconds * 1000
+
+		assert dateTime.isAfter(comparisonDateTime.minus(Duration.ofMillis(epsilonMilliseconds)))
+		assert dateTime.isBefore(comparisonDateTime.plus(Duration.ofMillis(epsilonMilliseconds)))
+	}
+
+	static void assertDateTimeFormat(dateTimeString)
+	{
+		assert dateTimeString ==~ /[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\.[0-9]{3}\+\d{4}/
+	}
+
+	static void assertEquals(String dateTimeString, LocalDate comparisonDate)
+	{
+		// Example date string: 2016-02-23
+		ZonedDateTime date = YonaServer.parseIsoDateString(dateTimeString)
+		assertEquals(date, comparisonDate)
+	}
+
+	static void assertEquals(LocalDate date, LocalDate comparisonDate)
+	{
+		assert date == comparisonDate
+	}
+
+	static void assertDateFormat(dateTimeString)
+	{
+		assert dateTimeString ==~ /[0-9]{4}-[0-9]{2}-[0-9]{2}/
+	}
+
+	static void assertResponseStatusCreated(def response)
+	{
+		assertResponseStatus(response, 201)
+	}
+
+	static void assertResponseStatus(def response, int status)
```

Comment:
Personally, I am not in favor of these methods. I prefer `response.status == 200` over `appService.assertResponseStatusSuccess(response)` and prefer `response.status == 301` over `appService.assertResponseStatus(response, 301)`.

Reply:
I understand, but see what these assert methods do: they show the error response of the server in the log message. That makes analysis of a failing test a lot simpler:

A regular ``assert response.status == 200`` gives this:
```
Condition not satisfied:

it.status == 200
|  |      |
|  400    false
groovyx.net.http.HttpResponseDecorator@5137927

	at nu.yona.server.UserTest.Try to create John Doe with an unsupported device operating system_closure1(UserTest.groovy:134)
	at nu.yona.server.test.AppService.addUser(AppService.groovy:68)
	at nu.yona.server.UserTest.Try to create John Doe with an unsupported device operating system(UserTest.groovy:137)
```

With ``AppService.assertResponseStatus(response, 200)``, you get this:
```
java.lang.AssertionError: Invalid status: 400 (expecting 200). Response: [message:Operating system 'RASPBIAN' is not supported., code:error.device.unknown.operating.system]. Expression: (response.status == status). Values: status = 200
	at nu.yona.server.test.AppService.assertResponseStatus(AppService.groovy:379)
	at nu.yona.server.UserTest.Try to create John Doe with an unsupported device operating system_closure1(UserTest.groovy:134)
	at nu.yona.server.test.AppService.addUser(AppService.groovy:68)
	at nu.yona.server.UserTest.Try to create John Doe with an unsupported device operating system(UserTest.groovy:137)
```

This makes it a lot quicker to find out what's going wrong.

Reply:
OK, that helps indeed. Have you replaced all response status assertions with those variants? Have you also created a `assertResponseStatusOK` that asserts on 200 instead of 2xx? I think that more precise asserts are preferable.

Reply:
Indeed, that's preferable. At the moment, it's done with the statement you see above. An ``assertResponseStatusOk`` would make it a little shorter. I propose to do that in another PR, as this is big enough already.

Reply:
I've created [YD-506](http://jira.yona.nu/browse/YD-506) for this and implemented it already, based on the change set that's in this PR.